src/queries/monitoring-grant-citation-report.sql

```diff
@@ -0,0 +1,140 @@
+/**
+* @name: Monitoring Grant Citation Report
+* @description: Retrieves monitoring report data based on various filters for compliance and audit purposes.
+* @defaultOutputName: monitoring_grant_citation_report
+* @technicalDescription: This query extracts monitoring report details including region, grant number, report delivery date, finding type, and citations, filtered by several parameters such as report delivery date, review outcomes, review types, finding reported date, and others.
+*
+* The query results are filterable by the following SSDI flags, which are passed as an array of values:
+* - ssdi.regionIds - integer[] - One or more values for 1 through 12
+* - ssdi.reportDeliveryDate - date[] - Two dates defining a range for the reportDeliveryDate to be within. If only one date is supplied, the range is from the supplied date to the current timestamp. If no dates are supplied, this filter is ignored.
+* - ssdi.reviewOutcomes - string[] - One or more review outcomes
+* - ssdi.reviewTypes - string[] - One or more review types
+* - ssdi.findingReportedDate - date[] - Two dates defining a range for the findingReportedDate to be within. If only one date is supplied, the range is from the supplied date to the current timestamp. If no dates are supplied, this filter is ignored.
+* - ssdi.findingStatuses - string[] - One or more finding statuses
+* - ssdi.findingTypes - string[] - One or more finding types
+* - ssdi.citations - string[] - One or more citations
+* - ssdi.grantNumbers - string[] - One or more grant numbers
+* - ssdi.recipients - string[] - One or more recipient names
+* - ssdi.uei - string[] - One or more UEI values
+*
+* Zero or more SSDI flags can be set within the same transaction as the query is executed.
+* The following is an example of how to set an SSDI flag:
+* SELECT SET_CONFIG('ssdi.reportDeliveryDate', '["2023-01-01", "2023-12-31"]', TRUE);
+*/
+SELECT DISTINCT
+    gr."regionId",
+    gr.number,
+    mr."reportDeliveryDate",
+    mr."reviewType",
+    mf."findingType",
+    ms."citation"
+FROM "Grants" gr
+JOIN "Recipients" r
+    ON gr."recipientId" = r.id
+JOIN "MonitoringReviewGrantees" mrg
+    ON gr.number = mrg."grantNumber"
+JOIN "MonitoringReviews" mr
+    ON mrg."reviewId" = mr."reviewId"
+JOIN "MonitoringReviewStatuses" mrs
+    ON mr."statusId" = mrs."statusId"
+JOIN "MonitoringFindingHistories" mfh
+    ON mr."reviewId" = mfh."reviewId"
+JOIN "MonitoringFindings" mf
+    ON mfh."findingId" = mf."findingId"
+JOIN "MonitoringFindingStandards" mfs
+    ON mf."findingId" = mfs."findingId"
+JOIN "MonitoringStandards" ms
+    ON mfs."standardId" = ms."standardId"
+JOIN "MonitoringFindingStatuses" mfs2
```

Comment:
I don’t know how I feel about all these being inner joins because it would make stuff silently drop out of the dataset if they ever uploaded dimensional tables without some status. That said, that seems like a very unlikely scenario.